src/qt/ee_debugwindow.cpp

```diff
@@ -0,0 +1,1231 @@
+    connect(memory_up, SIGNAL(clicked()), this, SLOT(on_memory_up_clicked()));
+    connect(memory_down, SIGNAL(clicked()), this, SLOT(on_memory_down_clicked()));
+    connect(memory_go_button, SIGNAL(clicked()), this, SLOT(on_memory_go_button_clicked()));
+    connect(memory_seek_ledit, SIGNAL(returnPressed()), this, SLOT(on_memory_seek_ledit_returnPressed()));
+
+    connect(break_continue_button, SIGNAL(clicked()), this, SLOT(on_break_continue_button_clicked()));
+    connect(step_button, SIGNAL(clicked()), this, SLOT(on_step_button_clicked()));
+    connect(next_button, SIGNAL(clicked()), this, SLOT(on_next_button_clicked()));
+    connect(finish_button, SIGNAL(clicked()), this, SLOT(on_finish_button_clicked()));
+    connect(add_breakpoint_combo, SIGNAL(activated(int )), this, SLOT(on_add_breakpoint_combo_activated(int)));
+    connect(cpu_combo, SIGNAL(activated(int)), this, SLOT(on_cpu_combo_activated(int)));
+    connect(breakpoint_table, SIGNAL(cellDoubleClicked(int, int)), this, SLOT(on_breakpoint_table_cellDoubleClicked(int, int)));
+}
+// set formatting for qtable (monospaced font, no borders)
+static void setup_qtable(QTableWidget* table, uint32_t size)
+{
+    table->verticalHeader()->hide();
+    table->horizontalHeader()->hide();
+    table->setShowGrid(false);
+
+    table->setEditTriggers(QAbstractItemView::NoEditTriggers); // prevent editing
+    table->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont)); // does this work on mac/windows?
+    table->setColumnCount(1);
+    table->setRowCount(size);
+    table->horizontalHeader()->setSectionResizeMode(QHeaderView::Stretch); // ??
+    for (uint32_t i = 0; i < size; i++)
+    {
+        table->setItem(i,0,new QTableWidgetItem("uninitialized"));
+        table->setRowHeight(i, 20);
+    }
+}
+
+// 128-bit register value to hex
+static QString reg_to_string(uint128_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X %08X %08X %08X", value._u32[3], value._u32[2], value._u32[1], value._u32[0]);
+    return QString(cstr);
+}
+
+static QString reg_to_string(uint64_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X %08X", (uint32_t)(value >> 32), (uint32_t)(value & 0xffffffff));
+    return QString(cstr);
+}
+
+static QString reg_to_string(uint32_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X", value);
+    return QString(cstr);
+}
+
+static QString reg_to_string(float value)
+{
+    char cstr[128];
+    sprintf(cstr, "%g", value);
+    return QString(cstr);
+}
+
+// word as hex, ascii, floating point
+static QString memory_data_to_string(uint32_t data)
+{
+    char* charp = (char*)(&data);
+
+    char ascii[5];
+
+    for (int j = 0; j < 4; j++)
+    {
+        if(charp[j] >= ' ' && charp[j] <= '~')
+            ascii[j] = charp[j];
+        else
+            ascii[j] = '.';
+    }
+    ascii[4] = '\0';
+    char fp[64];
+    sprintf(fp, "%g", *(float*)(&data));
+    return QString(reg_to_string(data) + "    " + QString(ascii) + "    " + QString(fp));
```

Comment:
Instead of using concatenation here, QString::arg can be used instead

```suggestion
    return QString("%1 %2 %3").arg(reg_to_string(data)).arg(ascii).arg(fp));
```